src/components/button/Button.tsx

```diff
@@ -0,0 +1,91 @@
+import styled from "@emotion/styled";
+import Image from "next/image";
+import { useState, ButtonHTMLAttributes } from "react";
+import { getButtonStyles, getWidthStyles, setLoadingStyles } from "components/button/styles";
+
+export interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
+  variant?: "primary" | "secondary" | "text" | "icon" | "iconWithText"; // color types
+  widthType?: "default" | "fullWidth"; // default : hug
+  isLoading?: boolean;
+  onClick?: () => void;
+}
+
+const Button = ({
+  children,
+  variant = "primary",
+  widthType = "default",
+  onClick,
+  ...props
+}: Props) => {
+  const [isButtonLoading, setIsButtonLoading] = useState(false);
+
+  const handleButtonClick = async () => {
+    setIsButtonLoading(true); // 클릭 시 loading 상태를 true로 변경
+
+    try {
+      if (onClick) {
+        await onClick(); // 부모 컴포넌트에서 전달받은 함수 호출
+      }
+    } catch (error) {
+      console.error("Error performing async work:", error);
+    }
+
+    setIsButtonLoading(false); // 완료 시 loading 상태를 false로 변경
+  };
+
+  return (
+    <EmotionWrapper
+      variant={variant}
+      widthType={widthType}
+      disabled={isButtonLoading}
+      onClick={handleButtonClick}
+      isLoading={isButtonLoading}
+      {...props}
+    >
+      {isButtonLoading && (
+        <Image src="/images/Spin-1s-100px-gray.svg" alt="Loading Spinner" width={14} height={14} />
+      )}
+      {children}
+    </EmotionWrapper>
+  );
+};
+
+export default Button;
+
+export const EmotionWrapper = styled.button<Props>`
+  padding: 4px 15px;
+  gap: 15px;
+  border-radius: 6px;
+  height: auto;
+
```

Comment:
```suggestion
  line-height: 1.5
```

가 추가되면 좋을 것 같습니다..! (디자인 안 기준)

현재 버튼이 디자인 보다 약간 납작(?) 한 느낌이 있어서요..!
![image](https://github.com/SystemConsultantGroup/foodhub-frontend/assets/86560973/799a878d-e8d3-4f70-a7cc-1bbe6862bf9f)

Reply:
컬러도 primary 일 경우 흰색 적용되어야 할 것 같습니다!

Reply:
이부분 해결된 것 같아 resolve 합니다~ 수고하셨습니다!